argument.go

```diff
@@ -121,6 +122,19 @@ func (o *arg) parse(args []string) error {
 	case *bool:
 		*o.result.(*bool) = true
 		o.parsed = true
+	case *int:
+		if len(args) < 1 {
+			return fmt.Errorf("[%s] must be followed by an integer", o.name())
+		}
+		if len(args) > 1 {
+			return fmt.Errorf("[%s] followed by too many arguments", o.name())
+		}
+		val, err := strconv.Atoi(args[0])
+		if err != nil {
+			return fmt.Errorf("[%s] must be followed by an integer", o.name())
```

Comment:
Hm, not sure whether error message in this case is good. The arg is already followed by something, which is not an integer, not sure if can improve this. Just something to think about.

Reply:
Another good point.  I used this wording since to get to this error, the value following the argument is not an integer otherwise `strconv.Atoi()` would not have returned an error...so the message is intended to emphasize that the parameter to the argument needs to be an _integer_.  I can't think of any improvements, though I'm very open to suggestions.